utils/s2n_init.c

```diff
@@ -49,6 +49,7 @@ int s2n_disable_atexit(void) {
 int s2n_init(void)
 {
     main_thread = pthread_self();
+    POSIX_GUARD_RESULT(s2n_libcrypto_init());
```

Comment:
Is this the right place in the init order? It's before we check that we haven't linked incorrectly (s2n_libcrypto_validate_runtime) and also before we set up locks, if that matters (s2n_locking_init). I've got a [PR](https://github.com/aws/s2n-tls/pull/3456) out with comments making the ordering logic more explicit, if you want to read those.

Reply:
yep just saw the PR https://github.com/aws/s2n-tls/pull/3456 and re-running tests to move it down the list

Reply:
also one of us is going to need a rebase so it should be handled at that point